### ISSUE TITLE
Deep link to Argo CD application lost with Promotion Directives

### ISSUE THREAD

**Issue by jessesuen** [akuity/kargo#2746]
# Description

We lost the deep link feature to the Argo CD application in v0.9. The following link is no longer present on Stages using promotion directives (but still present on ones that use promotionMechanisms).

![image](https://github.com/user-attachments/assets/8d2b3dec-754d-49bd-bad2-79b7f437e131)

# Steps to Reproduce

Any stage that references an Argo CD app using promotionTemplate steps will not have the link.

I think the UI will need to iterate the promotion steps and construct the links based on `argocd-update` steps.

```yaml
        - uses: argocd-update
          config:
            apps:
              - name: guestbook
```

# Version

```
v0.9
```

**Comment by krancour** [comment]
💯 

> I think the UI will need to iterate the promotion steps and construct the links based on argocd-update steps.

We've been trying hard to avoid most of Kargo having any specialized knowledge of the built-in steps so that the built-ins aren't any different from third-party ones, once that's enabled...

But that being said, I think this is one of a few issues that makes me pause and consider the possibility that built-ins might rightfully qualify as "special."

**Comment by jessesuen** [comment]
> We've been trying hard to avoid most of Kargo having any specialized knowledge of the built-in steps

I agree, and so I filed https://github.com/akuity/kargo/issues/2752 as a way to make built-ins less "special", or at least pave the way for 3rd party plugins to eventually have the ability display information in Kargo UI.

**Comment by Marvin9** [comment]
I included fix in https://github.com/akuity/kargo/pull/2694 which is available in v1.0.0. Closing it as of now. I agree that anything related to directives have to be pluggable, even for built-in ones it should be loosely coupled. As of now I kept all those related decision makings [traceable](https://github.com/akuity/kargo/blob/main/ui/src/features/promotion-directives/utils.ts#L4) in code
